lib/Conversion/FuseLinalg/FuseLinalg.cpp

```diff
@@ -35,8 +35,13 @@ class MatmulTranspose : public mlir::OpRewritePattern<mlir::linalg::MatmulOp> {
 
     for (size_t num_input = 0; num_input < inputs.size(); num_input++) {
       mlir::Value input = inputs[num_input];
-      transposeOp =
-          llvm::dyn_cast<mlir::linalg::TransposeOp>(input.getDefiningOp());
+      llvm::errs() << "input op: " << input.getDefiningOp() << "\n";
+      auto def_op = input.getDefiningOp();
+      // If one of inputs is input of whole func it will be nullptr.
+      if (!def_op) {
+        continue;
+      }
+      transposeOp = llvm::dyn_cast<mlir::linalg::TransposeOp>(def_op);
```

Comment:
also possible to fix with `llvm::dyn_cast_if_present`. I don’t know which option is better, it’s a stylistic choice.